purchases/src/main/kotlin/com/revenuecat/purchases/utils/LocaleExtensions.kt

```diff
@@ -0,0 +1,31 @@
+package com.revenuecat.purchases.utils
+
+import android.os.Build
+import androidx.annotation.RequiresApi
+import com.revenuecat.purchases.common.errorLog
+import java.util.Locale
+import java.util.MissingResourceException
+
+// This is a hack to make android studio previews work. In previews, the locale language method
+// gives a different result than expected (for example, "es-es" instead of "es").
+// In order to fix that, we convert the string to a locale and we parse that as a locale so the
+// language and country are correctly parsed.
+@RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+internal fun Locale.convertToCorrectlyFormattedLocale(): Locale {
+    return toString().toLocale()
+}
+
+@RequiresApi(Build.VERSION_CODES.LOLLIPOP)
```

Comment:
We have a util `Locale.toBCP47()` in `utils.kt` that we could use to make this also work pre-lollipop

Reply:
Nevermind, we have a min sdk of 24 in the UI module, which is where this is going to be used

Reply:
Oh yeah, I didn't know we had that. In any case, here we are converting a string to a locale, not the other way around, so we can't use that I think?